src/azure/storage/loader.rs

```diff
@@ -39,15 +40,33 @@ impl Loader {
         Ok(cred)
     }
 
+    /// Load credential with IMDS.
+    pub async fn load_with_imds(&self) -> Result<Option<Credential>> {
```

Comment:
We can’t add the ﻿load_imds logic to the existing ﻿load method here, because all the ﻿imds_* fields in Config are allowed to be empty. Introducing a new API allows us to make sure that the logic is handled properly.

Reply:
I don't want to expose a new API to users. It's Ok that all fields are empty after load.